exonum/src/runtime/rust/service.rs

```diff
@@ -83,21 +83,49 @@ where
     }
 }
 
+/// Transaction specification for the concrete service interface.
+pub trait Transaction: BinaryValue {
+    /// Service interface associated for the given transaction.
+    type Service;
+    /// Identifier of service method which executes the given transaction.
+    const METHOD_ID: MethodId;
+
+    /// Create unsigned service transaction from the value.
+    fn into_any_tx(self, instance_id: InstanceId) -> AnyTx {
+        AnyTx {
+            call_info: CallInfo {
+                instance_id,
+                method_id: Self::METHOD_ID,
+                // Only main interface can execute user transactions.
```

Comment:
Not sure whether we want to keep this restriction in the long run; I believe we've had the corresponding discussion internally.

Reply:
This is an obsolete comment.